src/cloud/mod.rs

```diff
@@ -0,0 +1,69 @@
+use crate::ops::ExitNode;
+use crate::ops::ExitNodeStatus;
+use async_trait::async_trait;
+use k8s_openapi::api::core::v1::Secret;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+pub mod aws;
+mod cloud_init;
+pub mod digitalocean;
+pub mod linode;
+pub mod pwgen;
+
+#[derive(Serialize, Deserialize, Debug, Clone, JsonSchema)]
+pub enum CloudProvider {
+    DigitalOcean,
+    Linode,
+    AWS,
+}
+
+// impl CloudProvider {
+//     #[allow(dead_code)]
+//     pub fn from_crd(crd: ExitNodeProvisioner) -> color_eyre::Result<Self> {
+//         // this thing is an enum so we can just match on it?
+
+//         // todo: single source of truth for this maybe
+//         // consider removing this CloudProvider enum and just using the ExitNodeProvisionerSpec enum
+//         // and then we can just match on that
+
+//         match crd.spec {
+//             ExitNodeProvisionerSpec::DigitalOcean(_) => Ok(CloudProvider::DigitalOcean),
+//             ExitNodeProvisionerSpec::Linode(_) => Ok(CloudProvider::Linode),
+//             ExitNodeProvisionerSpec::AWS(_) => Ok(CloudProvider::AWS),
+//         }
+//     }
+// }
+// This code was actually used, weirdly enough
+#[allow(dead_code)]
```

Comment:
um, okay.. we should fix this soon

Reply:
rust-analyzer complains because the crd generator doesnt use this part of code, but the main exec does